deluge/common.py

```diff
@@ -730,34 +730,41 @@ def is_ip(ip):
     """
     A simple test to see if 'ip' is valid
 
-    :param ip: the ip to check
-    :type ip: string
-    :returns: True or False
-    :rtype: bool
+    Args:
+        ip (str): The IP to test
+
+    Returns:
+        bool: Whether IP is valid is not
 
     ** Usage **
 
     >>> is_ip("127.0.0.1")
     True
 
     """
+    return is_ipv4(ip) or is_ipv6(ip)
+
+
+def is_ipv4(ip):
     import socket
-    # first we test ipv4
     try:
         if windows_check():
-            if socket.inet_aton(ip):
-                return True
+            return socket.inet_aton(ip)
         else:
             if socket.inet_pton(socket.AF_INET, ip):
                 return True
     except socket.error:
-        if not socket.has_ipv6:
-            return False
-    # now test ipv6
+        return False
+
+
+def is_ipv6(ip):
+    import socket
+    if not socket.has_ipv6:
+        return False
     try:
         if windows_check():
             log.warning("ipv6 check unavailable on windows")
-            return True
+            return None
```

Comment:
I don't think this should be `None` and it can't be `False` as we don't know so kind of leaves us back at `True`?